mars/services/scheduling/supervisor/assigner.py

```diff
@@ -164,7 +191,18 @@ async def assign_subtasks(self, subtasks: List[Subtask]):
                         max_size = size
                     elif size == max_size:
                         bands.append(band)
-            assigns.append(random.choice(bands))
+            band = random.choice(bands)
+            if band in exclude_bands and exclude_bands_force:
```

Comment:
Cheap eval first, exclude_bands_force and band in exclude_bands is better.

Reply:
Good catch